data.go

```diff
@@ -140,24 +140,48 @@ func (dbc *dbManager) getUnreadyBlobs() (ids []string, err error) {
 // TODO there's a bug in the db statement
 func (dbc *dbManager) getReadyDeployments() ([]DataDeployment, error) {
 
-	rows, err := dbc.getDb().Query(`SELECT
-	a.id,
-	a.organization_id,
-	a.environment_id,
-	a.bean_blob_id,
-	a.resource_blob_id,
-	a.type,
-	a.name,
-	a.revision,
-	a.path,
-	a.created_at,
-	a.created_by,
-	a.updated_at,
-	a.updated_by,
-	b.local_fs_location
-	FROM metadata_runtime_entity_metadata as a
-	INNER JOIN edgex_blob_available as b
-	ON (a.bean_blob_id = b.id OR a.resource_blob_id = b.id)
+	rows, err := dbc.getDb().Query(`
+		SELECT 	a.id,
+			a.organization_id,
+			a.environment_id,
+			a.bean_blob_id,
+			a.resource_blob_id,
+			a.type,
+			a.name,
+			a.revision,
+			a.path,
+			a.created_at,
+			a.created_by,
+			a.updated_at,
+			a.updated_by,
+			b.local_fs_location
+		FROM metadata_runtime_entity_metadata as a
+		INNER JOIN edgex_blob_available as b
+		ON (a.bean_blob_id = b.id)
+		WHERE a.id IN (
+			SELECT
+					a.id
+				FROM metadata_runtime_entity_metadata as a
+				INNER JOIN edgex_blob_available as b
+				ON a.resource_blob_id = b.id
+				WHERE a.resource_blob_id != ""
+			INTERSECT
+				SELECT
+					a.id
+				FROM metadata_runtime_entity_metadata as a
+				INNER JOIN edgex_blob_available as b
+				ON a.resource_blob_id = b.id
+				WHERE a.resource_blob_id != ""
```

Comment:
These two statements in "INTERSECT" are same. Anyway, I'll fix it in a later commit.